packages/x-charts-pro/src/internals/plugins/useChartProZoom/useChartProZoom.ts

```diff
@@ -44,6 +45,40 @@ export const useChartProZoom: ChartPlugin<UseChartProZoomSignature> = ({
   const drawingArea = useSelector(store, selectorChartDrawingArea);
   const optionsLookup = useSelector(store, selectorChartZoomOptionsLookup);
   const isZoomEnabled = Object.keys(optionsLookup).length > 0;
+  const onZoomChangeRef = React.useRef(0);
+
+  // Updating the `onZoomChange` prop is a common mistake that can lead to performance issues.
+  React.useEffect(() => {
```

Comment:
I wonder if we should wrap it `useEventCallback` instead? 

`onZoomChange` is an event callback, so it seems like a good solution. What do you think?